pulp_2to3_migration/tests/functional/test_migration_behaviour.py

```diff
@@ -0,0 +1,337 @@
+
+        Check correctness of the dara for the first repo in the list.
+        """
+        self._load_and_run(FILE_SIMPLE_PLAN)
+        pulp2repository = self.pulp2repositories_api.list(ordering='pulp2_id', limit=1).results[0]
+        pulp3_repo = self.file_repo_api.read(pulp2repository.pulp3_repository_href)
+        pulp3_remote = self.file_remotes_api.read(pulp2repository.pulp3_remote_href)
+        pulp3_pub = self.file_publications_api.read(pulp2repository.pulp3_publication_href)
+        pulp3_dist = self.file_distributions_api.read(pulp2repository.pulp3_distribution_hrefs[0])
+        remote_url = 'https://repos.fedorapeople.org/pulp/pulp/fixtures/file/PULP_MANIFEST'
+
+        self.assertEqual(len(self.pulp2repositories_api.list().results), 4)
+        self.assertTrue(pulp2repository.is_migrated)
+        self.assertEqual(pulp2repository.pulp2_repo_id, 'file')
+        self.assertEqual(pulp3_repo.name, 'file')
+        self.assertEqual(pulp3_repo.latest_version_href, pulp2repository.pulp3_repository_version)
+        self.assertEqual(pulp3_remote.url, remote_url)
+        self.assertEqual(pulp3_remote.policy, 'immediate')
+        self.assertEqual(pulp3_pub.manifest, 'PULP_MANIFEST')
+        self.assertEqual(pulp3_pub.repository_version, pulp2repository.pulp3_repository_version)
+        self.assertEqual(pulp3_pub.distributions[0], pulp3_dist.pulp_href)
+        self.assertEqual(pulp3_dist.base_path, 'file')
+
+    def test_pulp2content_simple_plan(self):
+        """Test pulp2content endpoint for a simple plan."""
+        self._test_pulp2content(FILE_SIMPLE_PLAN)
+
+    def test_pulp2content_complex_plan(self):
+        """Test pulp2content endpoint for a complex plan."""
+        self._test_pulp2content(FILE_COMPLEX_PLAN)
+
+    def test_pulp2repositories_simple_plan(self):
+        """Test pulp2repositories endpoint for a simple plan."""
+        self._test_pulp2repositories(FILE_SIMPLE_PLAN)
+
+    def test_pulp2repositories_complex_plan(self):
+        """Test pulp2repositories endpoint for a complex plan."""
+        self._test_pulp2repositories(FILE_COMPLEX_PLAN)
+
+    def test_many_into_one_repo(self):
+        """Test that many Pulp 2 repos can be migrated into one Pulp 3 repo."""
+        self._load_and_run(MANY_INTO_ONE_PLAN)
+        pulp2repositories = self.pulp2repositories_api.list().results
+        pulp3_repo = self.file_repo_api.list().results[0]
+        repo_content = self.file_content_api.list(repository_version=pulp3_repo.latest_version_href)
+
+        self.assertEqual(len(self.file_repo_api.list().results), 1)
+        self.assertEqual(self.file_repo_versions_api.list(pulp3_repo.pulp_href).count, 4)
+        for pulp2repo in pulp2repositories:
+            with self.subTest(pulp2repo=pulp2repo):
+                self.assertTrue(pulp2repo.is_migrated)
+                self.assertEqual(pulp2repo.pulp3_repository_href, pulp3_repo.pulp_href)
+        self.assertEqual(repo_content.count, 263)
```

Comment:
This is currently failing for a reason. I need to investigate it, looks like a bug at  first glance.